contentcuration/contentcuration/view/internal_views.py

```diff
@@ -42,18 +42,19 @@ def file_diff(request):
     data = json.loads(request.body)
 
     # Filter by file objects first to save on performance
-    difference = []
+    # difference = []
     to_return = []
-    in_db_list = File.objects.annotate(filename=Concat('checksum', Value('.'),  'file_format')).filter(filename__in=data).values_list('filename', flat=True)
+    # in_db_list = File.objects.annotate(filename=Concat('checksum', Value('.'),  'file_format')).filter(filename__in=data).values_list('filename', flat=True)
 
     # Double check that files are not corrupted
-    for f in in_db_list:
-        file_path = generate_file_on_disk_name(os.path.splitext(f)[0],f)
-        if not os.path.isfile(file_path):
-            difference += [f]
+    # for f in in_db_list:
+    #     file_path = generate_file_on_disk_name(os.path.splitext(f)[0],f)
+    #     if not os.path.isfile(file_path):
+    #         difference += [f]
 
     # Add files that don't exist in storage
-    for f in list(set(data) - set(difference)):
+    for f in data:
+        # for f in data:
         file_path = generate_file_on_disk_name(os.path.splitext(f)[0],f)
         # Write file if it doesn't already exist
         if not os.path.isfile(file_path) or os.path.getsize(file_path) == 0:
```

Comment:
Note (on the line below this) that it should be about 40% faster, when it's a single item, to do `to_return.append(f)` than `to_return += [f]`.